From ab3a7e47fa8218cf39578af1a4c8e4fda7eb0411 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lu=C3=ADs=20Pereira?= <luis.artur.pereira@gmail.com>
Date: Thu, 20 Apr 2017 11:11:58 +0100
Subject: [PATCH] Use const iterators (Fix a FTBFS in debug mode)

Avoid assignment of a non-const pointer to a const pointer and the
consequent container detach.

We now enforce it in LXQtCompilerSettings:
https://github.com/lxde/lxqt-build-tools/commit/8679fbacd09fb3afb316cd8d5ed19bc0d0ba6990
---
 pcmanfm/mainwindow.cpp        | 4 ++--
 pcmanfm/preferencesdialog.cpp | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/pcmanfm/mainwindow.cpp b/pcmanfm/mainwindow.cpp
index 60eff28..50c5714 100644
--- a/pcmanfm/mainwindow.cpp
+++ b/pcmanfm/mainwindow.cpp
@@ -984,8 +984,8 @@ void MainWindow::loadBookmarksMenu() {
 void MainWindow::onBookmarksChanged() {
     // delete existing items
     QList<QAction*> actions = ui.menu_Bookmarks->actions();
-    QList<QAction*>::const_iterator it = actions.begin();
-    QList<QAction*>::const_iterator last_it = actions.end() - 2;
+    QList<QAction*>::const_iterator it = actions.constBegin();
+    QList<QAction*>::const_iterator last_it = actions.constEnd() - 2;
 
     while(it != last_it) {
         QAction* action = *it;
diff --git a/pcmanfm/preferencesdialog.cpp b/pcmanfm/preferencesdialog.cpp
index f20c7e1..602b45c 100644
--- a/pcmanfm/preferencesdialog.cpp
+++ b/pcmanfm/preferencesdialog.cpp
@@ -91,7 +91,7 @@ void PreferencesDialog::initIconThemes(Settings& settings) {
 
         iconThemes.remove("hicolor"); // remove hicolor, which is only a fallback
         QHash<QString, QString>::const_iterator it;
-        for(it = iconThemes.begin(); it != iconThemes.end(); ++it) {
+        for(it = iconThemes.constBegin(); it != iconThemes.constEnd(); ++it) {
             ui.iconTheme->addItem(it.value(), it.key());
         }
         ui.iconTheme->model()->sort(0); // sort the list of icon theme names

